blueprint/core/blueprint_ImageView.h

```diff
@@ -81,20 +81,60 @@ namespace blueprint
         //==============================================================================
         juce::Image loadImageFromDataURL(juce::String& source)
         {
-            int index = source.indexOf(",");
-            juce::String base64Encoded = source.substring(index + 1);
+            // source is a data URL that describes image.
+            // the format is `data:[<mediatype>][;base64],<data>`
+            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
+            int commaIndex = source.indexOf(",");
+            int semiIndex = source.indexOf(";");
+            if (commaIndex == -1 || semiIndex == -1)
+            {
+                throw std::runtime_error("Image received an invalid data url.");
+            }
+            
+            auto base64EncodedData = source.substring(commaIndex + 1);
             juce::MemoryOutputStream outStream{};
 
-            if(!juce::Base64::convertFromBase64(outStream, base64Encoded))
+            if(!juce::Base64::convertFromBase64(outStream, base64EncodedData))
             {
-                throw std::logic_error("Image failed to convert data url.");
+                throw std::runtime_error("Image failed to convert data url.");
             }
 
-            std::unique_ptr<juce::InputStream> inputStream;
-            inputStream.reset(new juce::MemoryInputStream(outStream.getData(), outStream.getDataSize(), false));
+            juce::MemoryInputStream inputStream (outStream.getData(), outStream.getDataSize(), false);
+
+            auto mimeType = source.substring(5,semiIndex);
```

Comment:
So this will get a little tripped up I think. The `[;base64]` token is optional in a data URL if the representation of the data is textual. For us this almost doesn't matter, except that SVGs do indeed have textual representations: `data:image/svg+xml,<svg><path>etc</path></svg>`.

My current implementation just accepts raw svg strings and passes them to the Drawable below, but that's very unusual, and I suspect users will more likely expect to pass a proper data URL even for their SVGs. Really, my current approach takes advantage of the DOM hack that SVG is valid HTML :) So we should change that to be more in line with the spec.

So, generally, I think the approach here should be (ofc this is pseudo code because we don't have a URL class like this)
```
URL u(sourceString);
if (u.scheme() == "data:") { return handleDataURL(u); }
if (u.scheme() == "file:") { return handleFileURL(u); }
... same for http/https
```
Then `handleDataURL` should search for a `;base64` token. If it finds a `;base64`, remove it, and decode the remaining data string into a byte array. Then carry on to decoding based on the mime type as you've done here already.

That all said, I understand that the goal of this PR is to support jpeg/gif images in data URLs, and the intention was not to correct my existing SVG hack :D. And with that in mind, let's go ahead and merge this, and I'll open up a second issue to handle this step for SVG Data URLs and @zigen if you're interested in tackling it, that'd be great!

Reply:
thank you for reviewing!
I totally agree that `;base64` token handling. 
we should handle other types of base64 encoded data url.

I'm interested in it, let me try it later (perhaps in this month)

Reply:
Sounds great!